tests/unit/resources/test_compensation.py

```diff
@@ -98,3 +108,100 @@ def test_create_from_spill_string(spillstring):
+def test_apply_comp_errors_for_nonmatching_channels(
+    client, ENDPOINT_BASE, compensation, fcs_file
+):
+    events_body = open("tests/data/Acea - Novocyte.fcs", "rb")
+    responses.add(
+        responses.GET,
+        f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs",
+        body=events_body,
+    )
+    events = parse_fcs_file(client.download_fcs_file(EXP_ID, fcs_file._id))
+    fcs_file.events = events
+
+    with pytest.raises(IndexError):
+        compensation.apply(fcs_file)
+
+
+@responses.activate
+def test_apply_compensation_to_fcs_file_with_no_events(
+    client, ENDPOINT_BASE, compensation, fcs_file
+):
+    # Given: a Compensation with channels as a subset of the FcsFile events
+    events_body = open("tests/data/Acea - Novocyte.fcs", "rb")
+    responses.add(
+        responses.GET,
+        f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs",
+        body=events_body,
+    )
+    events = fcs_file.events
+
+    ix = list(events.columns)
+    compensation.dataframe = DataFrame(identity(24), index=ix, columns=ix)
+    compensation.channels = ix
+
+    # When: a Compensation is applied
+    results = compensation.apply(fcs_file)
+
+    # Then: events should be scaled
+    assert all(results == events)
+
+
+@responses.activate
+def test_apply_compensation_to_fcs_file_with_matching_kwargs(
+    client, ENDPOINT_BASE, compensation, fcs_file
+):
+    # Given: a Compensation with channels as a subset of the FcsFile events
+    responses.add(
+        responses.GET,
+        f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs",
+        body=open("tests/data/Acea - Novocyte.fcs", "rb"),
+    )
+    events = fcs_file.get_events(inplace=True, testKwarg="foo")
+    assert fcs_file._events_kwargs == {"testKwarg": "foo"}
+
+    ix = list(events.columns)
+    compensation.dataframe = DataFrame(identity(24), index=ix, columns=ix)
+    compensation.channels = ix
+
+    # When: a Compensation is applied
+    results = compensation.apply(fcs_file, testKwarg="foo")
+
+    # Then: events should be compensated
+    assert all(results == events)
+    assert (
+        responses.assert_call_count(
+            f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs?testKwarg=foo",
+            1,
+        )
+        is True
+    )
+
+
+@responses.activate
+def test_apply_comp_compensates_values(
+    acea_events_compensated,
+    acea_fcs_file,
+    acea_compensation,
+):
+    # Given:
+    # - a file-internal compensation (see tests/fixtures/compensated_events.py)
+    # - an FcsFile with uncompensated events
+
+    # When: the Compensation is applied to a file
+    results = acea_compensation.apply(acea_fcs_file, inplace=False)
+
+    # Then: events should be compensated correctly
+    assert all(
+        [
+            isclose(a, b)
+            for a, b in zip(
+                results.head(1).values.tolist()[0],
+                acea_events_compensated.head(1).values.tolist()[0],
```

Comment:
`head(1)` is selecting the first row, yeah? What's `[0]` selecting? (Not just the first column, right?)

If you have the entire 30 MB FCS file checked in, you might as well check a few more rows if it's just a matter of changing `head(1)` to `head(5)`.